### PR TITLE
feat: add staking for Kava & Stargaze

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/utils/network_info.ts [hunk 1]
```diff
@@ -408,6 +408,17 @@ const networks = {
     label: "Kava - KAVA",
     name: "Kava",
   },
+  "kava-testnet": {
+    address: "kavavaloper1vlpsrmdyuywvaqrv7rx6xga224sqfwz3yjnlkh",
+    delegate: "https://wallet.keplr.app/chains/kava", // FIXME test chain not found in keplr https://testnet.keplr.app/
```
Comment (icfor) [inline src/utils/network_info.ts:413]: I think could remove this line, this `delegate` property is a legacy feature and not important in new platform

File: src/utils/network_info.ts [hunk 2]
```diff
@@ -787,6 +809,7 @@ export const cosmosNetworkKeys = [
   "islamic_coin",
   "jackal",
   "kava",
+  "kava-testnet",
```
Comment (icfor) [inline src/utils/network_info.ts:812]: These properties are not filtered out in prod, so here they should be added dynamically (something like `.concat(ENABLE_TESTNETS ? ["kava-testnet", "stargaze-testnet"] : []) `)

File: src/utils/network_info.ts [hunk 3]
```diff
@@ -408,6 +408,17 @@ const networks = {
     label: "Kava - KAVA",
     name: "Kava",
   },
+  "kava-testnet": {
+    address: "kavavaloper1vlpsrmdyuywvaqrv7rx6xga224sqfwz3yjnlkh",
+    delegate: "https://wallet.keplr.app/chains/kava", // FIXME test chain not found in keplr https://testnet.keplr.app/
+    denom: "KAVA",
+    graphql: "kava",
+    guide: "how-to-stake-akt-on-kava",
+    image: "/images/network/kava.png",
```
Comment (icfor) [inline src/utils/network_info.ts:417]: This is fine, many images are png, although I've been moving some to svg, some logos will look a lot better, something to consider depending how they look

**Review by icfor — APPROVED** [review]

File: src/screens/staking/lib/staking_sdk/core.ts [hunk 4]
```diff
@@ -51,13 +57,18 @@ export const mainNetworkDenom: Record<StakingNetworkId, CoinDenom | null> = {
...
   [StakingNetworkId.Osmosis]: CoinDenom.OSMO,
   [StakingNetworkId.Regen]: CoinDenom.REGEN,
+  [StakingNetworkId.Stargaze]: CoinDenom.STARS,
+  [StakingNetworkId.StargazeTestnet]: CoinDenom.STARS,
 };
 
 export const testnetNetworks = new Set([
   StakingNetworkId.CosmosHubTestnet,
   StakingNetworkId.CelestiaTestnet,
+  StakingNetworkId.KavaTestnet,
```
Comment (icfor) [inline src/screens/staking/lib/staking_sdk/core.ts:71]: I think have to add stargaze testnet here

File: src/screens/staking/lib/staking_sdk/core.ts [hunk 5]
```diff
@@ -69,8 +80,12 @@ export const keplrNetworks = new Set(
     StakingNetworkId.CosmosHub,
     StakingNetworkId.CosmosHubTestnet,
     StakingNetworkId.DyDx,
+    StakingNetworkId.Kava,
+    StakingNetworkId.KavaTestnet,
     StakingNetworkId.Osmosis,
     StakingNetworkId.Regen,
+    StakingNetworkId.Stargaze,
+    StakingNetworkId.StargazeTestnet,
```
Comment (icfor) [inline src/screens/staking/lib/staking_sdk/core.ts:88]: Can't comment below but if I remember correctly, had to add both testnets to `keplrNonNativeChains`

File: src/utils/network_info.ts (hunk 1 above)
Comment (Juzorensen) [reply src/utils/network_info.ts]: You are right.
I forgot to push my changes 🙈.

File: src/screens/staking/lib/staking_sdk/core.ts (hunk 4 above)
Comment (Juzorensen) [reply src/screens/staking/lib/staking_sdk/core.ts:71]: Yup, related to the missing push 🙈

File: src/screens/staking/lib/staking_sdk/core.ts (hunk 5 above)
Comment (Juzorensen) [reply src/screens/staking/lib/staking_sdk/core.ts:89]: Yup, related to the missing push 🙈

File: src/utils/network_info.ts (hunk 3 above)
Comment (Juzorensen) [reply src/utils/network_info.ts:415]: Good to know.

I haven't added the asset though. I don't know from where to get the svg.

File: src/utils/network_info.ts (hunk 2 above)
Comment (Juzorensen) [reply src/utils/network_info.ts:814]: 4e0cfcb35391a0b091f6b34e3b33fc09f27c3bdc

File: src/utils/network_info.ts (hunk 3 above)
Comment (icfor) [reply src/utils/network_info.ts:417]: Ok, will send a DM with the figma project with logos

File: src/utils/network_info.ts [hunk 6]
```diff
@@ -810,6 +835,7 @@ export const cosmosNetworkKeys = [
   "ssv",
   "stafihub",
   "stargaze",
+  "stargaze-testnet",
```
Comment (icfor) [inline src/utils/network_info.ts:838]: Seems forgot to remove these

**Review by icfor — APPROVED** [review]

File: src/utils/network_info.ts (hunk 6 above)
Comment (Juzorensen) [reply src/utils/network_info.ts]: 🦅

Actually both of them 1783b42b0e6f883e20f5688fb85512ddb4ac80ab 🙈

File: src/utils/network_info.ts (hunk 3 above)
Comment (Juzorensen) [reply src/utils/network_info.ts]: 3e9852ab480bd5be4b5d9cab9053cb1695655518

**Pull request merged**